pos_accounting/src/pool/storage/mod.rs

```diff
@@ -184,7 +184,7 @@ impl<'a, S: PoSAccountingStorageWrite> PoSAccountingDBMut<'a, S> {
         })
     }
 
-    fn merge_data_generic<K: Ord + Copy, T: Clone, Getter, Setter, Deleter>(
+    fn merge_data_generic<K: Ord + Copy, T: Clone + PartialEq, Getter, Setter, Deleter>(
```

Comment:
Not sure `PartialEq` is completely well-behaved here as opposed to `Eq`. Have you considered the implications of it?